notebooks/catvton/ov_catvton_helper.py

```diff
@@ -0,0 +1,209 @@
+import os
+from collections import namedtuple
+from pathlib import Path
+import warnings
+
+from detectron2.export import TracingAdapter
+from detectron2.structures import Instances, Boxes
+from diffusers.image_processor import VaeImageProcessor
+from huggingface_hub import snapshot_download
+import yaml
+import openvino as ov
+import torch
+
+from model.cloth_masker import AutoMasker
+from model.pipeline import CatVTONPipeline
+
+
+def convert(model: torch.nn.Module, xml_path: str, example_input):
+    xml_path = Path(xml_path)
+    if not xml_path.exists():
+        xml_path.parent.mkdir(parents=True, exist_ok=True)
+        model.eval()
+        with torch.no_grad():
+            converted_model = ov.convert_model(model, example_input=example_input)
+        ov.save_model(converted_model, xml_path)
+
+        # cleanup memory
+        torch._C._jit_clear_class_registry()
+        torch.jit._recursive.concrete_type_store = torch.jit._recursive.ConcreteTypeStore()
+        torch.jit._state._clear_class_state()
+
+
+class VaeEncoder(torch.nn.Module):
+    def __init__(self, vae):
+        super().__init__()
+        self.vae = vae
+
+    def forward(self, x):
+        return self.vae.encode(x).latent_dist.sample()
```

Comment:
sample() includes random uniform that can make reproducing pytorch results numerical unstable. Possbly better to return distribution and implement sample in model inference wrapper.

Example how it can be achieved: 
conversion https://github.com/huggingface/optimum/blob/main/optimum/exporters/utils.py#L124
inference https://github.com/huggingface/optimum-intel/blob/main/optimum/intel/openvino/modeling_diffusion.py#L938

DiagonalGaussianDistribution is original vae encoder output that has sample() function